packages/react-material-ui/src/components/Table/README.md

```diff
@@ -0,0 +1,130 @@
+
+```js
+<Table.Root rows={customRows()} headers={headers}> // This works and the state is controlled by the component
+```
+
+```js
+// This also works and it means we're controlling the state outside of the component
+// If any of the additional props are passed, all others are required.
+<Table.Root 
+  rows={customRows()} 
+  headers={headers} 
+  tableQueryState={...} // I'm passing this, so every prop below are required
+  updateTableQueryState={...}
+  total={...}
+  pageCount={...}
+>
+```
+
+Instead of exposing a single Table component, we now expose each part of the Table component individually and integrate them using a custom hook called `useTableRoot` to share state.
+
+While we provide out-of-the-box components, we utilize render props in `Table.BodyRows`  to manage pagination, sorting, filtering, and more. However, we still grant developers the flexibility to use their own components for row rendering.
+
+```js
+<Table.BodyRows
+  renderRow={row => (
+    <Table.BodyRow row={row} hasCheckboxes={false} hover={false}>
+        <Table.BodyCell row={row} />
+    </Table.BodyRow>
+  )} 
+/>
+```
+
+You can create custom components and use the `useTableRoot` hook to access the necessary data from the Table state.
+
+```js
+const MyCustomComponent = () => {
+  const state = useTableRoot();
+
+  return (...)
+}
+```
+
+We now employ URL state for both manual pagination (server-side) and local pagination (client-side).
+We offer a function called createTableStyles to facilitate customizing the table layout without the need to create extensive objects directly in JSX/TSX. This enhances code readability and simplifies style toggling.
+
+```js
+const tableTheme = createTableStyles({
+  table: {
+    height: '100%',
+  },
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    flex: 1,
+    overflow: 'auto',
+  },
+
+<Table.Root sx={tableTheme.root}>
+<Table.Table stickyHeader variant="outlined" sx={tableTheme.table}>
+```
+
+## Hooks
+
+`useTable` now uses the `useTableQueryState` hook, creating a state object to manage pagination, filtering, and sorting. Importantly, `useTable` is now decoupled from the Table component and can function independently. It implements the same interface for the query state. based on the It can receive an initial state, or if one is not provided, it will utilize the default `useTableQueryState` settings. It exposes the fetched data, the current state, and provides methods for updating the state.
```

Comment:
"It implements the same interface for the query state. `based on the`"
Seems like this shouldn't be here